File: leetcode/2903/Solution.py
from typing import List

class Solution:
    def findIndices(self, nums: List[int], indexDifference: int, valueDifference: int) -> List[int]:
        n = len(nums)
        minIdx = 0
        maxIdx = 0
        for j in range(indexDifference, n):
            i = j - indexDifference
            if nums[i] < nums[minIdx]:
                minIdx = i
            elif nums[i] > nums[maxIdx]:
                maxIdx = i
            value = nums[j]
            if nums[maxIdx] - value >= valueDifference:
                return [maxIdx, j]
            if value - nums[minIdx] >= valueDifference:
                return [minIdx, j]
        return [-1, -1]

if __name__ == '__main__':
    sol = Solution()
    nums = [5,1,4,1]
    indexDifference = 2
    valueDifference = 4
    print("test:", sol.findIndices(nums, indexDifference, valueDifference))

File: leetcode/2903/Solution.java
import java.util.Arrays;

class Solution {
    public int[] findIndices(int[] nums, int indexDifference, int valueDifference) {
        int n = nums.length;
        int maxIdx = 0;
        int minIdx = 0;
        for (int j = indexDifference; j < n; j++) {
            int i = j - indexDifference;
            if (nums[i] < nums[minIdx]) {
                minIdx = i;
            } else if (nums[i] > nums[maxIdx]) {
                maxIdx = i;
            }
            int value = nums[j];
            if (nums[maxIdx] - value >= valueDifference) {
                return new int[]{maxIdx, j};
            }
            if (value - nums[minIdx] >= valueDifference) {
                return new int[]{minIdx, j};
            }
        }
        return new int[]{-1, -1};
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        int[] nums = {5,1,4,1};
        int indexDifference = 2;
        int valueDifference = 4;
        System.out.println("test: " + Arrays.toString(sol.findIndices(nums, indexDifference, valueDifference)));
    }
}